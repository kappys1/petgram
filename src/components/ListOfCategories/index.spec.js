// Category.test.js
/* global describe, it, expect afterAll jest */
import React from 'react'
// import renderer from 'react-test-renderer'
import { act, render } from '@testing-library/react'
import { ListOfCategories } from '.'

describe('ListOfCategories', () => {
  afterAll(() => {
    global.fetch.mockClear()
    delete global.fetch
  })
  it('snapshot', () => {
    const dummyMoviesData = [
      { id: '1', cover: 'some-1', path: '/', emoji: '' },
      { id: '1', cover: 'some-1', path: '/', emoji: '' },
      { id: '1', cover: 'some-1', path: '/', emoji: '' }
    ]
    global.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(dummyMoviesData)
      })
    )

    act(() => {
      const { asFragment } = render(<ListOfCategories />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
