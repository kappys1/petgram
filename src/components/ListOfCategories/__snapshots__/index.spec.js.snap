// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfCategories snapshot 1`] = `
<DocumentFragment>
  <ul
    class="sc-dlfnuX iqGdQY"
  />
</DocumentFragment>
`;
